packages/table/src/Table.tsx

```diff
@@ -14,19 +14,25 @@ const tableStyles = css`
   box-sizing: border-box;
 `;
 
-interface TableProps extends React.ComponentPropsWithoutRef<'table'> {
-  data?: Array<any>;
-  columns?:
+interface ChildrenArgInterface {
+  datum: any;
+  index: number;
+}
+export interface TableProps extends React.ComponentPropsWithoutRef<'table'> {
+  data: Array<any>;
+  columns:
     | Array<ReactElement<HeaderRowProps>>
     | Array<ReactElement<TableHeaderProps> | string>;
   selectable?: boolean;
+  children: (ChildArgObject: ChildrenArgInterface) => React.ReactElement;
 }
```

Comment:
I tried playing around with this and couldn't figure it out but it'd be awesome if we could do something like this:

```
interface ChildrenArgInterface<Shape> {
  datum: Shape;
  index: number;
}


export interface TableProps<Datum extends{} infer U ? U : any> extends React.ComponentPropsWithoutRef<'table'> {
  data: Array<U>;
  children: (ChildArgObject: ChildrenArgInterface<U>) => React.ReactElement;
}
```

Reply:
I think this works!!!
`type DataShape<Shape> = Shape extends infer U ? U : Shape;

interface TableRowInterface<Shape> {
  datum: DataShape<Shape>;
  index?: number;
}

export interface TableProps<Shape>
  extends React.ComponentPropsWithoutRef<'table'> {
  data: Array<DataShape<Shape>>;
  columns:
    | Array<ReactElement<HeaderRowProps>>
    | Array<ReactElement<TableHeaderProps> | string>
    | React.ReactFragment;
  selectable?: boolean;
  children: (TableRowArgs: TableRowInterface<Shape>) => JSX.Element;
}`